pypdf/generic/_annotations.py

```diff
@@ -25,6 +30,28 @@ def _get_bounding_rectangle(vertices: List[Tuple[float, float]]) -> RectangleObj
     return rect
 
 
+class AnnotationObject(DictionaryObject):
+    def set_border_color(
+        self,
+        border_color: Union[ArrayObject, List[float], Tuple[float, float, float], str],
+    ) -> "AnnotationObject":
+        """
+        Set the border color of a PDF annotation.
+
+        Args:
+            annotation: The PDF annotation to modify.
+            border_color: The color of the border in hex string format (e.g. "#FF0000" for red).
+        """
+        if isinstance(border_color, str):
+            if ImageColorLoaded is None:
+                raise ImportError("PIL required but not installed")
+            border_color = ImageColorLoaded.getrgb(border_color)
+        if isinstance(border_color, (list, tuple)):
+            border_color = ArrayObject([FloatObject(n) for n in border_color[:3]])
```

Comment:
```suggestion
            border_color = ArrayObject([FloatObject(n if n<=1.0 else n/255.0) for n in border_color[:3]])
```
if the array provided is an array of RGB color